packages/component_code_gen/templates/actions/additional_rules.py

```diff
@@ -0,0 +1,96 @@
+additional_rules = """## Additional rules
+
+When you generate code, you must follow all of the rules above. Review the rules and think through them step-by-step before you generate code. Look at how these map to the example code and component API described above.
```

Comment:
There are some rules here that should be shared across actions and sources. I think you could move #1 - #5, and #9 and #10 to a common file, then keep the other action-specific rules here.

Maybe it make sense to have the "common rules" start with the header `## Rules`, then call this header here `## Rules for writing actions` (same thing for the source rules)?